android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -1770,22 +1757,9 @@ public void setAdTagUrl(final Uri uri) {
         adTagUrl = uri;
     }
 
-    public void setRawSrc(final Uri uri, final String extension) {
-        if (uri != null) {
-            boolean isSourceEqual = uri.equals(srcUri);
-            this.srcUri = uri;
-            this.extension = extension;
-            this.mediaDataSourceFactory = buildDataSourceFactory(true);
-
-            if (!isSourceEqual) {
-                reloadSource();
-            }
-        }
-    }
-
     public void setTextTracks(SideLoadedTextTrackList textTracks) {
         this.textTracks = textTracks;
-        reloadSource();
+        reloadSource(); // FIXME Shall be moved inside source
```

Comment:
This fixme is for: https://github.com/TheWidlarzGroup/react-native-video/issues/3665

Reply:
nit;
I think we need to compare injected textTracks and this.textTracks before call reloadSource

Reply:
Yes and not. Yes you proposal would give a better behavior, but for exemple when we switch from a content with external Subtitle to a content with subtitle, the process is the following:
- we receive setSource of the new url
   -> we will reloadsource
- we receive an empty tracks
   -> we will anyway reload source.

As Exoplayer don't support dynamic external tracks update, I plan to move it into the source class.

I identified all following props which should be moved in source
https://github.com/TheWidlarzGroup/react-native-video/issues/2693#issuecomment-1291154625

Can be done without breaking change I think if the props are wrapped in JS part, but it will need update os ios and android implementation...
Notice that ios implementation is already prepared for this move with `struct VideoSource`